Traceback (most recent call last):
  File "/Users/hyosubkim/opt/anaconda3/envs/teaching/lib/python3.10/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/Users/hyosubkim/opt/anaconda3/envs/teaching/lib/python3.10/site-packages/nbclient/client.py", line 1314, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/Users/hyosubkim/opt/anaconda3/envs/teaching/lib/python3.10/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
  File "/Users/hyosubkim/opt/anaconda3/envs/teaching/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/Users/hyosubkim/opt/anaconda3/envs/teaching/lib/python3.10/site-packages/nbclient/client.py", line 709, in async_execute
    await self.async_execute_cell(
  File "/Users/hyosubkim/opt/anaconda3/envs/teaching/lib/python3.10/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Users/hyosubkim/opt/anaconda3/envs/teaching/lib/python3.10/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
df = pd.concat([pd.read_csv(f) for f in glob.glob('data/rl_s??_data.csv')],
              ignore_index=True)
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mValueError[0m                                Traceback (most recent call last)
Cell [0;32mIn[5], line 1[0m
[0;32m----> 1[0m df [38;5;241m=[39m [43mpd[49m[38;5;241;43m.[39;49m[43mconcat[49m[43m([49m[43m[[49m[43mpd[49m[38;5;241;43m.[39;49m[43mread_csv[49m[43m([49m[43mf[49m[43m)[49m[43m [49m[38;5;28;43;01mfor[39;49;00m[43m [49m[43mf[49m[43m [49m[38;5;129;43;01min[39;49;00m[43m [49m[43mglob[49m[38;5;241;43m.[39;49m[43mglob[49m[43m([49m[38;5;124;43m'[39;49m[38;5;124;43mdata/rl_s??_data.csv[39;49m[38;5;124;43m'[39;49m[43m)[49m[43m][49m[43m,[49m
[1;32m      2[0m [43m              [49m[43mignore_index[49m[38;5;241;43m=[39;49m[38;5;28;43;01mTrue[39;49;00m[43m)[49m

File [0;32m~/opt/anaconda3/envs/teaching/lib/python3.10/site-packages/pandas/core/reshape/concat.py:382[0m, in [0;36mconcat[0;34m(objs, axis, join, ignore_index, keys, levels, names, verify_integrity, sort, copy)[0m
[1;32m    379[0m [38;5;28;01melif[39;00m copy [38;5;129;01mand[39;00m using_copy_on_write():
[1;32m    380[0m     copy [38;5;241m=[39m [38;5;28;01mFalse[39;00m
[0;32m--> 382[0m op [38;5;241m=[39m [43m_Concatenator[49m[43m([49m
[1;32m    383[0m [43m    [49m[43mobjs[49m[43m,[49m
[1;32m    384[0m [43m    [49m[43maxis[49m[38;5;241;43m=[39;49m[43maxis[49m[43m,[49m
[1;32m    385[0m [43m    [49m[43mignore_index[49m[38;5;241;43m=[39;49m[43mignore_index[49m[43m,[49m
[1;32m    386[0m [43m    [49m[43mjoin[49m[38;5;241;43m=[39;49m[43mjoin[49m[43m,[49m
[1;32m    387[0m [43m    [49m[43mkeys[49m[38;5;241;43m=[39;49m[43mkeys[49m[43m,[49m
[1;32m    388[0m [43m    [49m[43mlevels[49m[38;5;241;43m=[39;49m[43mlevels[49m[43m,[49m
[1;32m    389[0m [43m    [49m[43mnames[49m[38;5;241;43m=[39;49m[43mnames[49m[43m,[49m
[1;32m    390[0m [43m    [49m[43mverify_integrity[49m[38;5;241;43m=[39;49m[43mverify_integrity[49m[43m,[49m
[1;32m    391[0m [43m    [49m[43mcopy[49m[38;5;241;43m=[39;49m[43mcopy[49m[43m,[49m
[1;32m    392[0m [43m    [49m[43msort[49m[38;5;241;43m=[39;49m[43msort[49m[43m,[49m
[1;32m    393[0m [43m[49m[43m)[49m
[1;32m    395[0m [38;5;28;01mreturn[39;00m op[38;5;241m.[39mget_result()

File [0;32m~/opt/anaconda3/envs/teaching/lib/python3.10/site-packages/pandas/core/reshape/concat.py:445[0m, in [0;36m_Concatenator.__init__[0;34m(self, objs, axis, join, keys, levels, names, ignore_index, verify_integrity, copy, sort)[0m
[1;32m    442[0m [38;5;28mself[39m[38;5;241m.[39mverify_integrity [38;5;241m=[39m verify_integrity
[1;32m    443[0m [38;5;28mself[39m[38;5;241m.[39mcopy [38;5;241m=[39m copy
[0;32m--> 445[0m objs, keys [38;5;241m=[39m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_clean_keys_and_objs[49m[43m([49m[43mobjs[49m[43m,[49m[43m [49m[43mkeys[49m[43m)[49m
[1;32m    447[0m [38;5;66;03m# figure out what our result ndim is going to be[39;00m
[1;32m    448[0m ndims [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39m_get_ndims(objs)

File [0;32m~/opt/anaconda3/envs/teaching/lib/python3.10/site-packages/pandas/core/reshape/concat.py:507[0m, in [0;36m_Concatenator._clean_keys_and_objs[0;34m(self, objs, keys)[0m
[1;32m    504[0m     objs_list [38;5;241m=[39m [38;5;28mlist[39m(objs)
[1;32m    506[0m [38;5;28;01mif[39;00m [38;5;28mlen[39m(objs_list) [38;5;241m==[39m [38;5;241m0[39m:
[0;32m--> 507[0m     [38;5;28;01mraise[39;00m [38;5;167;01mValueError[39;00m([38;5;124m"[39m[38;5;124mNo objects to concatenate[39m[38;5;124m"[39m)
[1;32m    509[0m [38;5;28;01mif[39;00m keys [38;5;129;01mis[39;00m [38;5;28;01mNone[39;00m:
[1;32m    510[0m     objs_list [38;5;241m=[39m [38;5;28mlist[39m(com[38;5;241m.[39mnot_none([38;5;241m*[39mobjs_list))

[0;31mValueError[0m: No objects to concatenate

